docs/how-to/hip_runtime_api/memory_management/unified_memory.rst

```diff
@@ -0,0 +1,737 @@
+and thereby reduced latency and bandwidth requirements. However, the actual
+impact on performance can vary based on the specific use case and the system.
+
+The following is the updated version of the example above with memory advice
+instead of prefetching.
+
+.. code-block:: cpp
+    :emphasize-lines: 29-41
+
+    #include <hip/hip_runtime.h>
+    #include <iostream>
+
+    #define HIP_CHECK(expression)              \
+    {                                          \
+        const hipError_t err = expression;     \
+        if(err != hipSuccess){                 \
+            std::cerr << "HIP error: "         \
+                << hipGetErrorString(err)      \
+                << " at " << __LINE__ << "\n"; \
+        }                                      \
+    }
+
+    // Addition of two values.
+    __global__ void add(int *a, int *b, int *c) {
+        *c = *a + *b;
+    }
+
+    int main() {
+        int deviceId;
+        HIP_CHECK(hipGetDevice(&deviceId));
+        int *a, *b, *c;
+
+        // Allocate memory for a, b, and c accessible to both device and host codes.
+        HIP_CHECK(hipMallocManaged(&a, sizeof(*a)));
+        HIP_CHECK(hipMallocManaged(&b, sizeof(*b)));
+        HIP_CHECK(hipMallocManaged(&c, sizeof(*c)));
+
+        // Set memory advice for a and b to be read, located on and accessed by the GPU.
+        HIP_CHECK(hipMemAdvise(a, sizeof(*a), hipMemAdviseSetPreferredLocation, deviceId));
+        HIP_CHECK(hipMemAdvise(a, sizeof(*a), hipMemAdviseSetAccessedBy, deviceId));
+        HIP_CHECK(hipMemAdvise(a, sizeof(*a), hipMemAdviseSetReadMostly, deviceId));
+
+        HIP_CHECK(hipMemAdvise(b, sizeof(*b), hipMemAdviseSetPreferredLocation, deviceId));
+        HIP_CHECK(hipMemAdvise(b, sizeof(*b), hipMemAdviseSetAccessedBy, deviceId));
+        HIP_CHECK(hipMemAdvise(b, sizeof(*b), hipMemAdviseSetReadMostly, deviceId));
+
+        // Set memory advice for c to be read, located on and accessed by the CPU.
+        HIP_CHECK(hipMemAdvise(c, sizeof(*c), hipMemAdviseSetPreferredLocation, hipCpuDeviceId));
+        HIP_CHECK(hipMemAdvise(c, sizeof(*c), hipMemAdviseSetAccessedBy, hipCpuDeviceId));
+        HIP_CHECK(hipMemAdvise(c, sizeof(*c), hipMemAdviseSetReadMostly, hipCpuDeviceId));
+
+        // Setup input values.
+        *a = 1;
+        *b = 2;
+
+        // Launch add() kernel on GPU.
+        hipLaunchKernelGGL(add, dim3(1), dim3(1), 0, 0, a, b, c);
+
+        // Wait for GPU to finish before accessing on host.
+        HIP_CHECK(hipDeviceSynchronize());
+
+        // Prints the result.
+        std::cout << *a << " + " << *b << " = " << *c << std::endl;
+
+        // Cleanup allocated memory.
+        HIP_CHECK(hipFree(a));
+        HIP_CHECK(hipFree(b));
+        HIP_CHECK(hipFree(c));
+
+        return 0;
+    }
+
+Memory range attributes
+--------------------------------------------------------------------------------
+
+:cpp:func:`hipMemRangeGetAttribute()` allows you to query attributes of a given
```

Comment:
This is a statement of what you can do, but it might be a good idea to say why you might do it or when you might do it?